force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,11 +185,61 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        
+        // Query to get the Account record matching the provided accountName
+        List<Account> accountList = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        
+        // Clone the list of opportunity names for further processing
+        List<String> oppNamesClone = oppNames.clone();
+
+        Account relatedAccount;
+    
+        // If the account does not exist, create a new Account record
+        if (accountList.isEmpty()) {
+            relatedAccount = new Account(Name = accountName);
+            insert relatedAccount;
+        } else {
+            // If the account exists, use the existing Account record
+            relatedAccount = accountList[0];
+        }
+
+        // Initialize a list to store the opportunities to be inserted/updated
+        List<Opportunity> opportunityList = new List<Opportunity>();
+
+        // Query to find existing opportunities that match the provided names
+        List<Opportunity> existingOppList = [SELECT Id, Name FROM Opportunity WHERE Name = :oppNames];
+
+        // If any existing opportunities are found, associate them with the related account
+        if (!(existingOppList.isEmpty())) {
```

Comment:
Some people claim this check is unnecessary because looping an empty list will loop 0 times. I personally do this check often because I feel like I like the mental check to process this block if not empty.